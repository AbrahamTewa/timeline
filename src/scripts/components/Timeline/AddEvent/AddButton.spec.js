/* eslint-env node, jest */

// ============================================================
// Import packages
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

// ============================================================
// Import modules
import AddButton from './AddButton';
import {
    snapshot,
} from '../../../testHelpers';

// ============================================================
// Tests

describe('Components', () => {
    describe('Unit test', () => {
        it('should react on changes', () => {
            const onClick = sinon.spy();
            const component = <AddButton onClick={onClick} />;
            const wrapper = shallow(component);
            wrapper.simulate('click');
            expect(onClick.calledOnce).toBe(true);
        });
    });

    describe('Snapshot', () => {
        it('should render normally', () => {
            snapshot(<AddButton onClick={() => {}} />);
        });
    });
});
