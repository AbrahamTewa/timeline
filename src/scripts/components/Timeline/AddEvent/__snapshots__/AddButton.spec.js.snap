// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Snapshot should render normally 1`] = `
<button
  className="btn btn-outline-primary btn-sm"
  onClick={[Function]}
  type="button"
>
  + Évènement
</button>
`;
